js-core-1/week-2/lesson.md

```diff
@@ -237,262 +271,135 @@ function satisfiesRequirements(num) {
 ```
 
-> When you call this array method it uses the array on the left side of the dot as an input, and it sorts that array also returning it. Note how both ordered and unordered arrays are sorted now!
-
-### `.concat()`
-
-_Adds (or concatenates) another value or array to the array._
+### Exercise (5 mins)
+In the node REPL, enter the following array:
 
 ```sh
-$ node
-> var arr = [1, 2, 3]
-undefined
-> arr.concat(4)
-[1, 2, 3, 4]
-> arr
-[1, 2, 3]
+> let fruits = ['banana', 'apple', 'strawberry', 'kiwi', 'fig', 'orange']
 ```
 
-Did you notice how calling the concat method did not change `arr`? This is because `concat`, like most array methods, returns a _new_ array, it does not alter the one you called the method on.
+1. Which index do you need to return:
+* strawberry
+* kiwi
+* orange
+* banana
 
-If you wan to use the array returned by calling `.concat()` you should store it in a new variable.
+2. Replace 'apple' with 'raspberry' and 'fig' with 'pineapple'
 
-```js
-var arr = [1, 2, 3];
-var newArr = arr.concat(4);
+### Exercise (5 mins)
+Complete this function so that, if the second index in array contains the name "Amy", it returns `"Second index matched!"`
 
-console.log(newArr); // logs [1, 2, 3, 4]
+```js
+function secondMatchesAmy(array) {
+  if ( ) {
+    return "Second index matched!"
+  }
+  
+  return "Second index not matched"
+}
 ```
 
-## More Array methods
-
-Let's explore some more array methods.
-
-### `.slice()`
-
-_Returns a slice of the array._
-
-You can tell `.slice()` where you want the slice to begin and end by passing it two parameters.
+## Loops
 
-```sh
-$ node
-> var arr = [0, 1, 2, 3, 4]
-undefined
-> arr.slice(0, 2)
-[0, 1]
-> ["a", "b", "c", "d"].slice(1, 2)
-['b']
-```
-
-### `.includes()`
+Loops go hand-in-hand with objects like arrays in JavaScript. A **loop**  is a way for you to go over each element in an array (or more generally, an **iterable** - we'll come to this later).
 
-_Returns true if a value is in the array._
+You can loop over a range of numbers (called the `for` statement in this course):
 
 ```js
-var mentors = ["Daniel", "Irini", "Ashleigh", "Rob", "Etzali"];
-
-function isAMentor(name) {
-  return mentors.includes(name);
+for (i=0; i<8; i++) {
+  console.log(i)
 }
-
-consooe.log("Is Rukmuni a mentor?");
-console.log(isAMentor("Rukmini")); // logs false
 ```
 
-### `.join()`
+Or over an array (called the `for/in` statement in this course):
 
-_Returns all the array values joined together in a string. By default, this method takes no parameters and then the elements are divided with a comma `,`. If you provide it with a string parameter though, then it becomes the divider of the elements, like the example below:_
+```js
+let fruits = ['banana', 'apple', 'strawberry', 'kiwi', 'fig', 'orange']
 
-```sh
-$ node
-> ["H", "e", "l", "l", "o"].join();
-'H,e,l,l,o'
-> ["H", "e", "l", "l", "o"].join("--");
-'H--e--l--l--o'
+for (i in fruits) {
+  console.log(i)
+}
```

Comment:
I think you are meant to use `for/of` here? `for/in` is to iterate through object properties. I wouldn't bother mentioning any of these constructions, to be honest (just `while` and `for`). Specially if we want to encourage students to use `map`, etc.

Reply:
I did test the code and worked, but I wasn't sure what benefits `for/of` provides over `for/in` with respect to arrays (I thought they behaved the same for arrays). Is there a difference?

Reply:
[for...in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) iterates over object properties, whereas [for...of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) is for iterable objects, like arrays. Because you're using `for...in` this logs the indexes (array keys) and not the values:

<img width="541" alt="Screenshot 2020-02-25 at 08 54 07" src="https://user-images.githubusercontent.com/2911171/75230956-8c6a7580-57ac-11ea-8e26-4fa694ef456f.png">

Reply:
Ah, thank you @ismaelgt ! I should question my memory more often :p